core/templates/cowdata.h

```diff
@@ -46,7 +46,7 @@ class CharString;
 template <class T, class V>
 class VMap;
 
-SAFE_NUMERIC_TYPE_PUN_GUARANTEES(uint64_t)
+static_assert(std::is_trivially_destructible<std::atomic<uint64_t>>::value);
```

Comment:
If this because of the `SafeNumeric<uint64_t>` around, I'd suggest including this in the set of type-pun guarantees macro in `safe_refcount.h`.

Reply:
It is part of `SAFE_NUMERIC_TYPE_PUN_GUARANTEES`:

```
#define SAFE_NUMERIC_TYPE_PUN_GUARANTEES(m_type)                    \
	static_assert(sizeof(SafeNumeric<m_type>) == sizeof(m_type));   \
	static_assert(alignof(SafeNumeric<m_type>) == alignof(m_type)); \
	static_assert(std::is_trivially_destructible<std::atomic<m_type>>::value);
```

We take case or sizes and alignment manually (since `alignof` part is not true on 32-bit platforms), and I have copy-pasted the remaining check here (but I'm not sure if there's any point for checking it for `uint64_t` at all).

Reply:
Ah, makes sense.